prudp_endpoint.go

```diff
@@ -527,6 +527,7 @@ func (pep *PRUDPEndPoint) handleReliable(packet PRUDPPacketInterface) {
 	connection := packet.Sender().(*PRUDPConnection)
 
 	slidingWindow := packet.Sender().(*PRUDPConnection).SlidingWindow(packet.SubstreamID())
+	slidingWindow.Lock()
```

Comment:
Alternativelly we can do:

```suggestion
	slidingWindow.Lock()
	defer slidingWindow.Unlock()
```

This will make sure that the mutex is unlocked before returning in the case that we add an early return in the future